public/src/components/dashboard/SubmissionsTable.tsx

```diff
@@ -35,7 +35,8 @@ const SubmissionsTable = (): JSX.Element => {
             }
             // Submissions are indexed by the assignment order - 1.
             const submission = submissions[assignment.order - 1] ?? new Submission()
-            if (submission.status !== Submission_Status.APPROVED && assignment.deadline) {
+            const status = getStatusByUser(submission, state.self.ID)
+            if (status !== Submission_Status.APPROVED && assignment.deadline) {
```

Comment:
Perhaps we could use a helper method here as well (to check for approval)? (See other comment about the same in another file.)